service/worker/scheduler/workflow.go

```diff
@@ -336,6 +355,43 @@ func (s *scheduler) processPatch(patch *schedpb.SchedulePatch) {
 	}
 }
 
+func (s *scheduler) timeGenerator() func(time.Time) getNextTimeResult {
+	var nextResults [TimeGeneratorCacheSize]getNextTimeResult
+	currentIndex := 0
+	lastCacheElement := -1
+	return func(after time.Time) getNextTimeResult {
```

Comment:
I think this might be easier to read if we just put these fields in the `scheduler` and made this a regular function instead of a closure?

Also, if `nextResults` is a slice then you don't need to keep the two indexes, the slice len is enough (strip off the first one each time you use it). So I think the cache can just be one field?

Reply:
good points and SGTM.